modules/widget-visibility/widget-conditions.php

```diff
@@ -560,7 +574,7 @@ public static function filter_widget( $instance ) {
 								$condition_result = is_archive();
 							break;
 							case 'posts':
-								$condition_result = $wp_query->is_posts_page;
+								$condition_result = is_single();
```

Comment:
This is a definite change -- `is_posts_page` returns true for the posts archive view when there's a page on the site's home.  `is_single()` returns true on a page's permalink, non-archive view.

https://github.com/WordPress/WordPress/blob/9d7ea04936125041b224b227f48ad4194066b314/wp-includes/class-wp-query.php#L1008